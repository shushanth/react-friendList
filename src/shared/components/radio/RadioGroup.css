:local(.radioGroup) {
  display: inherit;
  width: 100%;
}
