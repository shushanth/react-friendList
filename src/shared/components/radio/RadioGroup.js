/**
 * Generic radio group component
 */

import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

import styles from "./RadioGroup.css";

const RadioGroup = props => {
  const radioGroups = React.Children.map(props.children, child => {
    if (child.type.name === "RadioOption") {
      return React.cloneElement(child, {
        isChecked: props.radioValue === child.props.value,
        onChange: props.onRadioChange,
        name
      });
    }
  });
  return <div className={classnames(styles.radioGroup)}>{radioGroups}</div>;
};

RadioGroup.PropTypes = {
  name: PropTypes.string,
  radioValue: PropTypes.string,
  onRadioChange: PropTypes.func
};

export default RadioGroup;
