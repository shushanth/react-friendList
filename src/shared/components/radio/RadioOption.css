:local(.radioOption) {
  align-items: center;
  cursor: pointer;
  display: inherit;
  font-family: inherit;
  font-size: inherit;
}

:local(.radioOption input) {
  margin: 0;
}

:local(.radioOption label) {
  color: #fff;
  cursor: pointer;
  font-weight: 400;
  line-height: 2;
  letter-spacing: 0.1em;
  margin: 0 0.5em -0.1em;
  padding: 0;
}
