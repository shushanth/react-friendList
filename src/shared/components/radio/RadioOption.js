/**
 * Generic component for radio option, used in composed component (Radio group)
 */

import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

import styles from "./RadioOption.css";

const RadioOption = ({ name, isChecked, value, onChange }) => {
  return (
    <div className={classnames(styles.radioOption)}>
      <input
        className={classnames(styles.radioOption)}
        type="radio"
        value={value}
        id={value}
        name={name}
        checked={isChecked}
        onChange={onChange}
      />
      <label htmlFor={value}>{value}</label>
    </div>
  );
};

RadioOption.PropTypes = {
  name: PropTypes.string,
  checked: PropTypes.bool,
  onChange: PropTypes.func
};

export default RadioOption;
