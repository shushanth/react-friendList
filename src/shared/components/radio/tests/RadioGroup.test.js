import React from "react";
import { mount, shallow } from "enzyme";

import RadioGroup from "../RadioGroup";
import RadioOption from "../RadioOption";

describe("Component: <RadioGroup/>", () => {
  let RadioGroupWrapper;
  let RadioGroupMountWrapper;
  let props;
  beforeAll(() => {
    props = {
      name: "gender",
      radioValue: "Female",
      onRadioChange: jest.fn()
    };
    RadioGroupWrapper = shallow(
      <RadioGroup {...props}>
        <RadioOption value="Male" />
        <RadioOption value="Female" />
      </RadioGroup>
    );
    RadioGroupMountWrapper = mount(
      <RadioGroup {...props}>
        <RadioOption value="Male" />
        <RadioOption value="Female" />
      </RadioGroup>
    );
  });

  it("should render without crashing", () => {
    expect(RadioGroupWrapper.exists()).toBe(true);
    expect(RadioGroupWrapper).toMatchSnapshot();
    expect(RadioGroupWrapper.find("RadioOption").length).toEqual(2);
  });

  it("should not render radio option, when RadioOption is not present", () => {
    let radioGroupWithOption = shallow(<RadioGroup {...props} />);
    expect(radioGroupWithOption.find("div").children().length).toBe(0);
  });

  it("should have the radioOption based on childrens of the same", () => {
    let radioGroupOneOption = shallow(
      <RadioGroup {...props}>
        <RadioOption value="Male" />
      </RadioGroup>
    );
    expect(radioGroupOneOption.find("RadioOption").length).toEqual(1);
  });

  it("should have radioOption value same as prop value", () => {
    expect(
      RadioGroupWrapper.find("RadioOption")
        .at(0)
        .props().value
    ).toEqual("Male");
    expect(
      RadioGroupWrapper.find("RadioOption")
        .at(1)
        .props().value
    ).toEqual(props.radioValue);
  });

  it("should have radioOption isChecked boolean value, same as prop radioValue", () => {
    let radioGroupMount = shallow(
      <RadioGroup {...{ radioValue: "Male" }}>
        <RadioOption value="Male" />
        <RadioOption value="Female" />
      </RadioGroup>
    );
    expect(
      RadioGroupWrapper.find("RadioOption")
        .at(0)
        .props().isChecked
    ).toBeFalsy();
    expect(
      RadioGroupWrapper.find("RadioOption")
        .at(1)
        .props().isChecked
    ).toBeTruthy();
    expect(
      radioGroupMount
        .find("RadioOption")
        .at(0)
        .props().isChecked
    ).toBeTruthy();
    expect(
      radioGroupMount
        .find("RadioOption")
        .at(1)
        .props().isChecked
    ).toBeFalsy();
  });
  describe("method: onRadioChange", () => {
    it("should invoke props onRadioChange method on change in radioOption", () => {
      let inputRadioOption = RadioGroupMountWrapper.find("RadioOption")
        .at(0)
        .find("input");
      inputRadioOption.simulate("change");
      expect(props.onRadioChange).toHaveBeenCalled();
    });
  });
});
