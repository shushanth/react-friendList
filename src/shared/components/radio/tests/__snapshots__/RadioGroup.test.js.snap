exports[`Component: <RadioGroup/> should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RadioGroup
    name="gender"
    onRadioChange={[Function]}
    radioValue="Female">
    <RadioOption
        value="Male" />
    <RadioOption
        value="Female" />
</RadioGroup>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <RadioOption
          isChecked={false}
          name="nodejs"
          onChange={[Function]}
          value="Male" />,
        <RadioOption
          isChecked={true}
          name="nodejs"
          onChange={[Function]}
          value="Female" />,
      ],
      "className": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": ".0",
        "nodeType": "function",
        "props": Object {
          "isChecked": false,
          "name": "nodejs",
          "onChange": [Function],
          "value": "Male",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": ".1",
        "nodeType": "function",
        "props": Object {
          "isChecked": true,
          "name": "nodejs",
          "onChange": [Function],
          "value": "Female",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <RadioOption
            isChecked={false}
            name="nodejs"
            onChange={[Function]}
            value="Male" />,
          <RadioOption
            isChecked={true}
            name="nodejs"
            onChange={[Function]}
            value="Female" />,
        ],
        "className": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": ".0",
          "nodeType": "function",
          "props": Object {
            "isChecked": false,
            "name": "nodejs",
            "onChange": [Function],
            "value": "Male",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": ".1",
          "nodeType": "function",
          "props": Object {
            "isChecked": true,
            "name": "nodejs",
            "onChange": [Function],
            "value": "Female",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
